// jsdocument/objects.js
function CreateHTMLObject() {
    return {Text:"HTMLObject", Type:"div",ObjectType:"JSDocument",Arguments:""}
}

var DefaultObject = CreateHTMLObject()

const HTMLObject = (function() {
    function HTMLObject() {
        this.Text = "HTMLObject"
        this.ObjectType = "JSDocument"
        this.Arguments = ""
        this.Type = "div"
    }

    function SetType(newtype) {
        this.Type = newtype
    }
    HTMLObject.prototype.SetType = SetType

    return HTMLObject;
}())

const HTMLScriptObject = (function() {
    function HTMLScriptObject() {
        this.Text = "HTMLObject"
        this.ObjectType = "JSDocument"
        this.Arguments = ""
        this.Type = "script"
    }

    return HTMLScriptObject
}())

const HTMLStyleObject = (function() {
    function HTMLStyleObject() {
        this.Text = ""
        this.ObjectType = "JSDocument"
        this.Arguments = ""
        this.Type = "style"
    }

    return HTMLStyleObject
}())

const HTMLLinkObject = (function() {
    function HTMLLinkObject(href, rel) {
        this.Text = ""
        this.ObjectType = "JSDocument"
        this.Arguments = `href=${href} rel=${rel}`
        this.Type = "link"
    }

    return HTMLLinkObject
}())

const HTMLH1Object = (function() {
    function HTMLH1Object() {
        this.Text = "HTMLObject"
        this.ObjectType = "JSDocument"
        this.Arguments = ""
        this.Type = "h1"
    }

    return HTMLH1Object
}())

const HTMLH2Object = (function() {
    function HTMLH2Object() {
        this.Text = "HTMLObject"
        this.ObjectType = "JSDocument"
        this.Arguments = ""
        this.Type = "h2"
    }

    return HTMLH2Object
}())

const HTMLH3Object = (function() {
    function HTMLH3Object() {
        this.Text = "HTMLObject"
        this.ObjectType = "JSDocument"
        this.Arguments = ""
        this.Type = "h3"
    }

    return HTMLH3Object
}())

const HTMLH4Object = (function() {
    function HTMLH4Object() {
        this.Text = "HTMLObject"
        this.ObjectType = "JSDocument"
        this.Arguments = ""
        this.Type = "h4"
    }

    return HTMLH4Object
}())

const HTMLH5Object = (function() {
    function HTMLH5Object() {
        this.Text = "HTMLObject"
        this.ObjectType = "JSDocument"
        this.Arguments = ""
        this.Type = "h5"
    }

    return HTMLH5Object
}())

const HTMLVideoObject = (function() {
    function HTMLVideoObject(src) {
        this.Text = "HTML5 Video not supported on your device."
        this.ObjectType = "JSDocument"
        this.Arguments = `src=${src}`
        this.Type = "video"
    }

    return HTMLVideoObject
}())

const HTMLPObject = (function() {
    function HTMLPObject(src) {
        this.Text = "HTMLObject"
        this.ObjectType = "JSDocument"
        this.Arguments = ""
        this.Type = "p"
    }

    return HTMLPObject
}())

const HTMLTitleObject = (function() {
    function HTMLTitleObject() {
        this.Text = "HTMLObject"
        this.ObjectType = "JSDocument"
        this.Arguments = ""
        this.Type = "title"
    }

    function SetTitle(newtitle) {
        this.Text = newtitle
    }
    HTMLTitleObject.prototype.SetTitle = SetTitle

    return HTMLTitleObject;
}())

module.exports = {JSObject:HTMLObject, JSTitleObject:HTMLTitleObject, JSScriptObject:HTMLScriptObject, JSStyleObject:HTMLStyleObject, JSLinkObject:HTMLLinkObject,
                  JSH1Object:HTMLH1Object, JSH2Object:HTMLH2Object, JSH3Object:HTMLH3Object, JSH4Object:HTMLH4Object, JSH5Object:HTMLH5Object, JSVideoObject:HTMLVideoObject,
                  JSPObject:HTMLPObject}
